### PR TITLE
[feat]: [PIE-4858]: Validate overlay input sets on input set update

### PR DESCRIPTION
**Pull request by namanvermah**
If pipeline update makes an input set invalid, all overlay input sets referring to it are made invalid. After this, if a user updates an invalid input set, any overlay input set that refers to that input set needs to be validated again.
Also, if an input set is deleted, then all overlay input sets that refer to that input set need to be marked as invalid.
This is only for Inline cases.

This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36579)

### CODE REVIEW

File: pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/helpers/InputSetErrorsHelper.java [hunk 1]
```diff
@@ -117,6 +117,26 @@ public Map<String, String> getInvalidInputSetReferences(
     return res;
   }
 
+  public Map<String, String> getInvalidInputSetReferences(
```
Comment (namanvermah) [inline pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/helpers/InputSetErrorsHelper.java:120]: Need to change the method name here.

File: pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/OverlayInputSetValidationHelper.java [hunk 2]
```diff
@@ -75,6 +76,38 @@ public void validateOverlayInputSet(
     }
   }
 
+  public void validateOverlayInputSetsForGivenInputSet(
```
Comment (namanvermah) [inline pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/OverlayInputSetValidationHelper.java:79]: Add test for this (and all new methods actually)

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 1 Issues 
* Code issue 1 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/36579)

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 1 of 3 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36579)* status: 6 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/OverlayInputSetValidationHelper.java` line 84 at r2](https://reviewable.io/reviews/harness/harness-core/36579#-NA3RYbt853P4ORKypFe:-NA3RYbt853P4ORKypFf:b-gw7el9) ([raw file](https://github.com/harness/harness-core/blob/1bf7c0162e3df67b5a6a7d2ccb2263701fe1eed1/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/OverlayInputSetValidationHelper.java#L84)):*
> ```Java
>     if (validatedInputSet.getStoreType() != StoreType.INLINE
>         || validatedInputSet.getInputSetEntityType() != InputSetEntityType.INPUT_SET) {
>       return;
> ```

can we add comment to handle validate for overlay set for remote?



_Code quote:_
```Java
return;
```

**Review by namanvermah — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36579)* status: 6 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/OverlayInputSetValidationHelper.java` line 84 at r2](https://reviewable.io/reviews/harness/harness-core/36579#-NA3RYbt853P4ORKypFe:-NA3UtDT69NDClu8jC7d:b-w1o0cj) ([raw file](https://github.com/harness/harness-core/blob/1bf7c0162e3df67b5a6a7d2ccb2263701fe1eed1/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/OverlayInputSetValidationHelper.java#L84)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

can we add comment to handle validate for overlay set for remote?
</blockquote></details>

This can never be handled for remote. For remote we don’t do any validations.

**Review by archit-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36579)* status: 6 file(s) to review. 1 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/OverlayInputSetValidationHelper.java` line 84 at r2](https://reviewable.io/reviews/harness/harness-core/36579#-NA3RYbt853P4ORKypFe:-NA3VnDK7u76lFcKqZ7V:b-d2m7tv) ([raw file](https://github.com/harness/harness-core/blob/1bf7c0162e3df67b5a6a7d2ccb2263701fe1eed1/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/service/OverlayInputSetValidationHelper.java#L84)):*
<details><summary><i>Previously, namanvermah (Naman Verma) wrote…</i></summary><blockquote>

This can never be handled for remote. For remote we don’t do any validations.
</blockquote></details>

discussed offline

**Review by prashantsharma04 — APPROVED** [review]
> Reviewed 1 of 3 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36579)* status: :shipit: complete! ✓

**Pull request merged**